### ISSUE TITLE
Bug in "Edit on Github" links?

### ISSUE THREAD

**Issue by aallan** [raspberrypi/documentation#2855]
In the [camera software page](https://www.raspberrypi.com/documentation/computers/camera_software.html) there is a potential bug showing up in the "Edit this on Github" links.

This section, the "[libcamera and libcamera apps](https://www.raspberrypi.com/documentation/computers/camera_software.html#libcamera-and-libcamera-apps)" section (which corresponds to `camera/libcamera_apps_intro.adoc`) has a "Edit this on Github" link. 

<img width="556" alt="Screenshot 2023-03-09 at 09 19 18" src="https://user-images.githubusercontent.com/53625/223982243-af5ca42f-4d03-4cdf-b9df-874808d99181.png">

However, the following "[Getting Started](https://www.raspberrypi.com/documentation/computers/camera_software.html#getting-started)" section title does not have an "Edit this on Github" link, despite being a separate file (`camera/libcamera_apps_getting_started.adoc`).

<img width="484" alt="Screenshot 2023-03-09 at 09 19 10" src="https://user-images.githubusercontent.com/53625/223982335-ee26b419-65a1-4be3-b72c-50bb57a56d04.png">

Now the "Getting Started" section is one level down in header than the previous section, but I thought the critera for an edit link was whether there was an associated asciidoc file with that header title (rather than it was a top level H block?). 

Top level include file (`camera.adoc`) looks like this,

````
include::camera/libcamera_apps_intro.adoc[]

include::camera/libcamera_apps_getting_started.adoc[]
````

Is this a corner case where we weren't expecting an included file to exist as a lower level heading item inside a section header where that's an entirely different file.

**Comment by aallan** [comment]
This is (only) a potential bug because while I think everything that was an asciidoc file was supposed to have an "Edit this in Github" I'm not sure whether we'd decided on every top level section title instead? 🤔

**Comment by nelliemckesson** [comment]
So, from what I'm seeing in [create_build_adoc.py](https://github.com/raspberrypi/documentation/blob/develop/scripts/create_build_adoc.py#L66), it looks like the GithubEdit links are indeed only being added for top-level (`==`) headers. @lurch can you confirm?

I think we can shift stuff around to add them for all adoc files, but I want to give Andrew a chance to chime in first...

**Comment by aallan** [comment]
Honestly can't remember whether this was a deliberate decision on our (my!?) part? @lurch can you remember? I guess I'm actually a little bit torn as to whether we should change it. But I'm coming down on the "if it's a separate file it should get an 'Edit on Github' link" side of things, I do actually use then (and I thought I wouldn't!).

**Comment by aallan** [comment]
Still a bug.

**Comment by nelliemckesson** [comment]
I just made @lurch 's recommended fix to #2990 - take a look and let me know if that's what you're after!

**Comment by aallan** [comment]
That seemed to fix it!
